lib/learn_open/adapters/learn_web_adapter.rb

```diff
@@ -0,0 +1,101 @@
+module LearnOpen
+  module Adapters
+    class LearnWebAdapter
+      attr_reader :client
+      def initialize(options={})
+        @client = options.fetch(:learn_web_client) { LearnOpen.learn_web_client }
+      end
+
+      def fetch_lesson_data(target_lesson: false, fetch_next_lesson: false)
+        if opening_current_lesson?(target_lesson, fetch_next_lesson)
+          load_current_lesson
+          {
+            lesson: current_lesson,
+            id: current_lesson.id,
+            later_lesson: false
+          }
+        elsif opening_next_lesson?(target_lesson, fetch_next_lesson)
+          load_next_lesson
+          {
+            lesson: next_lesson,
+            id: next_lesson.id,
+            later_lesson: false
+          }
+        else
+          lesson = correct_lesson(target_lesson)
```

Comment:
`correct_lesson` here is a specific requested lesson that's not the current or next lesson, yeah? like when you try to open a lab that's beyond your current lesson?

thinking about the name (again, i'm here for only the most super critical important stuff)... "correct" is kind of a heavy term... lots of connotations, might be easy to misunderstand. like at first read, i thought maybe we had some elaborate method for determining the "correct" lesson a user should be on, based on track hierarchy or something.

i wonder if it'd be any clearer to rename to `later_lesson` to match the key referenced below.

Reply:
I thought `correct_lesson` was to be read as "correcting the lesson"...https://github.com/learn-co/learn-open/pull/17#discussion_r194858058. 

Take the target lesson in, validate it, return the right one. "correct" seems like the right name, but it still throw me off.

Reply:
Word. That's a hold over from the previous implementation. I'm going to go with something like `get_lesson(target_lesson, retries=3)`